appnav/src/main/kotlin/io/element/android/appnav/LoggedInFlowNode.kt

```diff
@@ -471,17 +478,31 @@ class LoggedInFlowNode @AssistedInject constructor(
         waitForNavTargetAttached { navTarget ->
             navTarget is NavTarget.RoomList
         }
-        attachChild<RoomFlowNode> {
-            backstack.push(
-                NavTarget.Room(
-                    roomIdOrAlias = roomIdOrAlias,
-                    serverNames = serverNames,
-                    trigger = trigger,
-                    initialElement = RoomNavigationTarget.Messages(
-                        focusedEventId = eventId
+
+        val isRoomAlreadyDisplayed = backstack.active?.key?.navTarget?.let { target ->
+            target is NavTarget.Room && target.roomIdOrAlias == roomIdOrAlias
+        } == true
+
+        if (isRoomAlreadyDisplayed) {
```

Comment:
Maybe we should always replace the backstack with RoomList+Room no matter what is currently in the backstack

Reply:
exemple : if you are following some permalinks, you can have several rooms in the backstack, if you click on a notif I think we want the behavior to always be the same